src/molecules/BottomSheet/BottomSheet.native.js

```diff
@@ -65,123 +65,142 @@ const BottomSheetDragBar = styled(View)`
   border-radius: ${(props) => props.theme.spacings.xsmall};
 `;
 
-const BottomSheet = forwardRef(
-  (
-    {
-      snapPoint = DEFAULT_SNAP_POINT,
-      children,
-      onBackDropClick = () => {},
-      onChange = () => {},
-      onClose = () => {},
-      adjustToContentHeight = false,
-      initialHeight = 0,
-    },
-    ref,
-  ) => {
-    const theme = useTheme();
-    const [headerHeight, setHeaderHeight] = useState(0);
-    const [contentHeight, setContentHeight] = useState(0);
-    const bottomsheetChildrenGroupByType = reactChildrenGroupByType(children);
-
-    const headerComponent = bottomsheetChildrenGroupByType[BottomSheetHeader];
-    const footerComponent = bottomsheetChildrenGroupByType[BottomSheetFooter];
-    const contentComponent = bottomsheetChildrenGroupByType[BottomSheetContent];
-
-    if (headerComponent?.length > 1) {
-      throw new Error(
-        `expected to have single \`BottomSheet.Header\` but found ${headerComponent.length}`,
-      );
-    }
+const BottomSheet = ({
+  visible = false,
+  snapPoint = DEFAULT_SNAP_POINT,
+  children,
+  onBackDropClick = () => {},
+  onChange = () => {},
+  onClose,
+  adjustToContentHeight = false,
+  initialHeight = 0,
+}) => {
+  const theme = useTheme();
+  const bottomSheetRef = useRef();
+  const [headerHeight, setHeaderHeight] = useState(0);
+  const [contentHeight, setContentHeight] = useState(0);
+  const bottomsheetChildrenGroupByType = reactChildrenGroupByType(children);
+
+  const headerComponent = bottomsheetChildrenGroupByType[BottomSheetHeader];
+  const footerComponent = bottomsheetChildrenGroupByType[BottomSheetFooter];
+  const contentComponent = bottomsheetChildrenGroupByType[BottomSheetContent];
 
-    if (footerComponent?.length > 1) {
-      throw new Error(
-        `expected to have single \`BottomSheet.Footer\` but found ${footerComponent.length}`,
-      );
+  useEffect(() => {
+    if (visible) {
+      bottomSheetRef.current?.open();
+    } else {
+      bottomSheetRef.current?.close();
     }
+  }, [visible]);
 
-    const handleHeaderLayoutChange = useCallback((e) => {
-      setHeaderHeight(e.nativeEvent.layout.height);
-    }, []);
+  if (!onClose) {
+    throw new Error(`expected to provide an onClose method for \`BottomSheet\``);
+  }
 
-    const handleContentLayoutChange = useCallback((e) => {
-      setContentHeight(e.nativeEvent.layout.height);
-    }, []);
+  if (headerComponent?.length > 1) {
+    throw new Error(
```

Comment:
don't think we need to provide `new` keyword. Can't see in checkbox or other places